src/FS.Physical/PhysicalFilesWatcher.cs

```diff
@@ -339,26 +374,56 @@ private static bool IsDirectoryPath(string path)
                 path[path.Length - 1] == Path.AltDirectorySeparatorChar);
         }
 
-        private static void CancelToken(ChangeTokenInfo matchInfo)
+        private static Task CancelToken(CancellationTokenSource cancellationTokenSource)
         {
-            if (matchInfo.TokenSource.IsCancellationRequested)
+            if (cancellationTokenSource.IsCancellationRequested)
             {
-                return;
+                return Task.CompletedTask;
             }
 
-            Task.Run(() =>
+            return Task.Run(() =>
```

Comment:
Avoid the closure 😄

Reply:
I'll undo the change. That said, not super sure how I'd avoid the closure here. Need to pass in a cts to a background Task

Reply:
Like this https://github.com/aspnet/Caching/blob/ee13d634568fc5348e09fc090fab40233b1107cc/src/Microsoft.Extensions.Caching.Memory/MemoryCache.cs#L297